src/backend/PortabilityService.ConfigurationSource/ConfigurationSource.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Extensions.Configuration.Memory;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace PortabilityService.ConfigurationSource
+{
+    public class PortabilityServiceConfigurationSource : MemoryConfigurationSource
+    {
+        private const string DefaultConfigurationServiceUrl = "http://localhost:1443";
+        internal const string PortabilityServiceConfigurationRoot = "PortabilityServiceSettings";
+
+        public PortabilityServiceConfigurationSource(string configurationSection = PortabilityServiceConfigurationRoot)
+            : this(GetConfigurationServiceUrlFromEnvironment() ?? new Uri(DefaultConfigurationServiceUrl), configurationSection) { }
+
+        public PortabilityServiceConfigurationSource(Uri configurationServiceUrl, string configurationSection = PortabilityServiceConfigurationRoot)
+            : this(CreateHttpClient(configurationServiceUrl ?? throw new ArgumentNullException(nameof(configurationServiceUrl))), configurationSection ?? throw new ArgumentNullException(nameof(configurationSection))) { }
+
+        public PortabilityServiceConfigurationSource(HttpClient httpClient, string configurationSection = PortabilityServiceConfigurationRoot)
+        {
+            InitialData = GetConfigurationSectionSettings(configurationSection ?? throw new ArgumentNullException(nameof(configurationSection)), httpClient ?? throw new ArgumentNullException(nameof(httpClient))).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Gets the Portability configuration service URL from the environment
+        /// </summary>
+        /// <returns></returns>
+        private static Uri GetConfigurationServiceUrlFromEnvironment()
+        {
+            var urlFromEnvironment = Environment.GetEnvironmentVariable("PortabilityConfigurationServiceUrl");
+
+            if (string.IsNullOrEmpty(urlFromEnvironment))
+            {
+                return null;
+            }
+            else
+            {
+                return new Uri(urlFromEnvironment);
+            }
+        }
+
+        /// <summary>
+        /// Creates an HTTPClient with the appsettings.json Url
+        /// </summary>
+        private static HttpClient CreateHttpClient(Uri configurationServiceUrl) => new HttpClient() { BaseAddress = configurationServiceUrl };
+
+        /// <summary>
+        /// Gets the configuration settings for the passed in section
+        /// </summary>
+        private static async Task<IEnumerable<KeyValuePair<string, string>>> GetConfigurationSectionSettings(string configurationSection, HttpClient httpClient)
+            => await GetJsonDataAsync<IEnumerable<KeyValuePair<string, string>>>($"/api/Configuration/sectionsettingslist/{configurationSection}", httpClient);
```

Comment:
I would put this into a `Constants.cs` class.

Reply:
This should probably be in a config provider (rather than a source) to enable reloading, etc.

Reply:
Also, have we tested whether the /sectionsettingslist endpoint properly handle arrays of settings? For example, with [this sample json configuration](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/configuration/index/sample/ConfigJson/appsettings.json) will we correctly receive both wizards? It seems worth testing since the idea of [] config values may not map properly to an `IEnumerable<KVP>`. 

This will be important for PortabilityService since Serilog sinks are provided as an [] of settings, as well as Ocelot routing information (though the latter might not be in shared configuration).

Reply:
We have a test that checks that the sectionsettingslist endpoint handles arrays in the ConfigurationService tests.